### PR TITLE
[fix](point query) Fix NegativeArraySizeException when prepared state…

### PR DESCRIPTION
**Pull request by mymeiyi**
…ment contains a long string

# Proposed changes

Issue Number: close #xxx

## Problem summary

When use prepared statement with a long string, get a NegativeArraySizeException in client side, the exception in fe is:

```
2023-03-10 14:03:39,527 WARN (mysql-nio-pool-1|645) [ConnectProcessor.handleExecute():249] Process one query failed because unknown reason: 
java.lang.NegativeArraySizeException: null
        at org.apache.doris.analysis.StringLiteral.setupParamFromBinary(StringLiteral.java:308) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.handleExecute(ConnectProcessor.java:235) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.dispatch(ConnectProcessor.java:566) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.processOnce(ConnectProcessor.java:794) ~[doris-fe.jar:1.2-SNAPSHOT]
        at org.apache.doris.mysql.ReadListener.lambda$handleEvent$0(ReadListener.java:52) ~[doris-fe.jar:1.2-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_131]
        at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_131]
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LiteralExpr.java [hunk 1]
```diff
@@ -334,41 +335,36 @@ public static int getParmLen(ByteBuffer data) {
...
+            In our client-server protocol all numbers bigger than 2^24
+            stored as 8 bytes with uint8korr. Here we always know that
+            parameter length is less than 2^4 so we don't look at the second
+            4 bytes. But still we need to obey the protocol hence 9 in the
+            assignment below.
+            */
+            if (maxLen < 5) {
+                return 0;
+            }
+            return MysqlProto.readInt4(data);
```
Comment (eldenmoon) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LiteralExpr.java:362]: is it enough to just read 4 bytes? but mysql protocol consists of 8 bytes when len = 254

**Comment by eldenmoon** [comment]
run buildall

File: regression-test/suites/point_query_p0/test_point_query.groovy [hunk 2]
```diff
@@ -69,6 +78,9 @@ suite("test_point_query") {
       sql """ INSERT INTO ${tableName} VALUES(1235, 991129292901.11138, "dd", null, "2120-01-02", "2020-01-01 12:36:38", 652.692, "5022-01-01 11:30:38") """
       sql """ INSERT INTO ${tableName} VALUES(1236, 100320.11139, "laa    ddd", "laooq", "2220-01-02", "2020-01-01 12:36:38", 2.7692, "6022-01-01 11:30:38") """
       sql """ INSERT INTO ${tableName} VALUES(1237, 120939.11130, "a    ddd", "laooq", "2030-01-02", "2020-01-01 12:36:38", 22.822, "7022-01-01 11:30:38") """
+      sql """ INSERT INTO ${tableName} VALUES(251, 120939.11130, "${generateString(251)}", "laooq", "2030-01-02", "2020-01-01 12:36:38", 251, "7022-01-01 11:30:38") """
```
Comment (eldenmoon) [inline regression-test/suites/point_query_p0/test_point_query.groovy:81]: INSERT INTO is not prepared, so the case is not effect the FE code modified. For now Literal.getParmLen only for SelectStatement

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.11 seconds
 stream load tsv:          469 seconds loaded 74807831229 Bytes, about 152 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          75 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          33 seconds loaded 861443392 Bytes, about 24 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230310085540_clickbench_pr_112139.html

File: regression-test/suites/point_query_p0/test_point_query.groovy (hunk 2 above)
Comment (mymeiyi) [reply regression-test/suites/point_query_p0/test_point_query.groovy:81]: Yes, add this in order to check select result.

**Review by eldenmoon — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LiteralExpr.java (hunk 1 above)
Comment (mymeiyi) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/LiteralExpr.java]: fixed

**Comment by eldenmoon** [comment]
run buildall

**Comment by mymeiyi** [comment]
run buildall

**Comment by mymeiyi** [comment]
run feut

**Comment by mymeiyi** [comment]
run p0

**Comment by mymeiyi** [comment]
run clickbench

**Comment by mymeiyi** [comment]
run p0

**Comment by mymeiyi** [comment]
run p1

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**